src/unix/thread.c

```diff
@@ -51,6 +51,60 @@ sysreturn arch_prctl(int code, unsigned long addr)
 }
 #endif
 
+struct clone_args_internal {
+     u64 flags;
+     int *child_tid;
+     int *parent_tid;
+     void *stack;
+     bytes stack_size;
+     u64 tls;
+};
+
+sysreturn clone_internal(struct clone_args_internal *args, bytes size)
+{
+     u64 flags = args->flags;
+
+     if (!args->stack_size)
+          return set_syscall_error(current, EINVAL);
+
+     if (!(flags & CLONE_THREAD)) {
+          thread_log(current, "attempted to create new process, aborting.");
+          return set_syscall_error(current, ENOSYS);
```

Comment:
I know these calls to set_syscall_error() are in the current code, but they are actually unnecessary, because these are direct returns (i.e. the syscall handler returns immediately without blocking). I would replace `set_syscall_error(current, <errno>)` with `-<errno>`; e.g. the above line becomes `return -ENOSYS;`